xml/value.go

```diff
@@ -0,0 +1,339 @@
+// It will auto close the xml element tag.
+func (xv Value) BigInteger(v *big.Int) {
+	writeStartElement(xv.w, xv.startElement)
+	xv.w.Write([]byte(v.Text(10)))
+	writeEndElement(xv.w, xv.endElement)
+}
+
+// BigDecimal encodes v big.Float as XML value.
+// It will auto close the xml element tag.
+func (xv Value) BigDecimal(v *big.Float) {
+	if i, accuracy := v.Int64(); accuracy == big.Exact {
+		xv.Long(i)
+		return
+	}
+
+	writeStartElement(xv.w, xv.startElement)
+	xv.w.Write([]byte(v.Text('e', -1)))
+	writeEndElement(xv.w, xv.endElement)
+}
+
+// Null encodes a null element tag like <root></root>.
+// It will auto close the xml element tag.
+func (xv Value) Null() {
+	// write open tag for the parent object
+	writeStartElement(xv.w, xv.startElement)
+
+	// close tag
+	writeEndElement(xv.w, xv.endElement)
+}
+
+// Write writes v directly to the xml document
+// if escapeXMLText is set to true, write will escape text.
+// It will auto close the xml element tag.
+func (xv Value) Write(v []byte, escapeXMLText bool) {
+	writeStartElement(xv.w, xv.startElement)
+
+	// escape and write xml text
+	if escapeXMLText {
+		escapeText(xv.w, v)
+	} else {
+		// write xml directly
+		xv.w.Write(v)
+	}
+
+	writeEndElement(xv.w, xv.endElement)
+}
+
+// NestedElement returns a nested element encoding.
+// It returns a point to element object and a close function that will
+// close the element tag.
+func (xv Value) NestedElement() (o *Object) {
+	// write open tag for the parent object
+	writeStartElement(xv.w, xv.startElement)
+
+	return newObject(xv.w, xv.scratch, xv.endElement)
+}
+
+// Array returns an array encoder. By default, the members of array are
+// wrapped with `<member>` element tag.
+func (xv Value) Array() *Array {
+	// write start element for the array element
+	writeStartElement(xv.w, xv.startElement)
+
+	return newArray(xv.w, xv.scratch, xv.endElement, arrayMemberWrapper)
+}
+
+/*
+ArrayWithCustomName returns an array encoder.
+
+It takes name as an argument, the name will used to wrap xml array entries.
+for eg, <someList><customName>entry1</customName></someList>
+Here `customName` element tag will be wrapped on each array member.
+*/
+func (xv Value) ArrayWithCustomName(name string) (a *Array) {
+	// write open tag for the array element
+	writeStartElement(xv.w, xv.startElement)
+
+	return newArray(xv.w, xv.scratch, xv.endElement, name)
+}
+
+/*
+FlattenedArray returns a flattened array encoder.
+
+FlattenedArray Encoder wraps each member with array's root element tag, thus
+flattening the array.
+
+for eg,`<someList>entry1</someList><someList>entry2</someList>`.
+*/
+func (xv Value) FlattenedArray() (a *Array) {
+	return newFlattenedArray(xv.w, xv.scratch, xv.startElement, xv.endElement)
+}
+
+/*
+Map returns a map encoder. By default, the map entries are
+wrapped with `<entry>` element tag.
+
+for eg. <entry><k>entry1</k><v>value1</v></entry><entry><k>entry2</k><v>value2</v></entry>
+*/
+func (xv Value) Map() *Map {
```

Comment:
Nit this example for `map` should include the wrapping tag caused by the `Value`

```xml
<somemap><entry>...</entry></somemap>
```